src/pages/workspaces/workspace/WorkspaceContainer.js

```diff
@@ -247,8 +246,10 @@ export const wrapWorkspace = ({ breadcrumbs, activeTab, title, topBarContent, sh
       cachedWorkspace :
       undefined
     const [googleProject, setGoogleProject] = useState(workspace?.workspace.googleProject)
-    const [location, setLocation] = useState(defaultLocation)
-    const [locationType, setLocationType] = useState(locationTypes.default)
+
+    const cachedLocation = Utils.useStore(workspaceBucketLocationStore)
+    const location = cachedLocation?.location
+    const locationType = cachedLocation?.locationType
```

Comment:
Is it possible for `location` or `locationType` to be undefined? If it is, what would the downstream implication be?

Reply:
Yes. Downstream components would need to fall back to the defaultLocation and defaultLocationType.

Reply:
Saw some opportunity to remove state around bucket location and made a PR into your branch thinking it might be easier to explain rather than in comments. Let me know what you think.

Reply:
The main change in that PR seems to undo the bug you discovered and I fixed in [this comment](https://github.com/DataBiosphere/terra-ui/pull/2680#issuecomment-973167744). See my comment that PR for details.